### PR TITLE
Integrate SelectCommand and PolicyList into GUI

### PR DESCRIPTION
**Pull request by tituswe**
### Select core functionality complete, integrated into GUI.

Selecting will allow users to highlight a client. Client's then information will be presented on the Policy Panel of the GUI.

**How To Use:** enter `select <index>` on the CLI

### Changes to CommandResult and Command classes

Do take note that the `CommandResult` class has be refactored to contain the `targetClient` as a proprietary field.
This `targetClient` attribute allows for the `Logic` component to access the selected client to display on the GUI.
This added feature can be taken advantage of for further implementations of new features.

Please do test out the feature and update me if there are any bugs encountered. Cheers!

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 1]
```diff
@@ -4,38 +4,83 @@
...
 
     private final String feedbackToUser;
 
-    /** Help information should be shown to the user. */
+    private final Client targetClient;
+
+    /**
+     * Selects a particular client
+     */
+    private final boolean select;
```
Comment (anunayajoshi) [inline src/main/java/seedu/address/logic/commands/CommandResult.java:21]: Do you think we should rename the boolean to isSelected, to stick to coding standards?

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 2]
```diff
@@ -4,38 +4,83 @@
...
+    /**
+     * Help information should be shown to the user.
+     */
     private final boolean showHelp;
 
-    /** The application should exit. */
+    /**
+     * The application should exit.
+     */
     private final boolean exit;
```
Comment (anunayajoshi) [inline src/main/java/seedu/address/logic/commands/CommandResult.java:31]: Same case here for the boolean coding standard.

**Review by anunayajoshi — APPROVED** [review]
Good job writing new test cases! LGTM!

File: src/main/java/seedu/address/logic/commands/CommandResult.java (hunk 1 above)
Comment (tituswe) [reply src/main/java/seedu/address/logic/commands/CommandResult.java:21]: Ooh, great idea!

**Review by pangrwa — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/CommandResult.java (hunk 1 above)
Comment (tituswe) [reply src/main/java/seedu/address/logic/commands/CommandResult.java:21]: `select` is not used in the conventional sense here, it follows the same convention as the other two boolean state variables `showHelp` and `exit`. 

It is supposed to represent the state of the command being entered.

**Pull request merged**